boot.js

```diff
@@ -28,30 +28,31 @@ module.exports = function (cb) {
 
   // 3. Load conf-sample.js and merge
   var defaults = require('./conf-sample')
-  _.defaultsDeep(conf, config, defaults)
```

Comment:
imho redundant as `--conf` already takes precedence over `conf.js`

1. `conf` gets filled by `--conf` (the object)
2. `config` (content of `conf.js`) and `defaults` (content of `conf-sample.js`) sources fill `conf` with all new properties

The priority should always be `--conf -> conf.js -> conf-sample.js` 

We call `--conf` config **overrides** for a reason.

Reply:
Indeed, you're correct. I misdiagnosed a different issue for this. However, I'm still willing to submit this as-is because `_.defaultsDeep()` mutates the source object, which in this case is the object from the `require()` cache and I think it's a bad practice to modify that object since it gets `require()`d elsewhere in the app and that code expects it to be unmodified.

Reply:
I initially had `_.defaultsDeep(conf, ...` as an empty object and sourced it using `overrides, config, defaults)` in yesterdays PR but reworked it as it looked like an overkill.

> mutates the source object

According to the docs this only mutates the object (not the sources) which i am fine this as this is indeed our running config. The require alone wouldn't suffice anyway, unless someone parses an entire working config using `--conf` of course.

Reply:
By mutating the object (which is what is going to be returned on subsequent `require()` calls to the override path), we end up providing a full copy of the config to the commands that process the overrides for their internal `s.options` i.e.: https://github.com/DeviaVir/zenbot/blob/unstable/commands/sim.js#L52 - instead of just the subset in the overrides file, which will be a much larger set of properties than it currently retains.

Reply:
I actually prefer the idea of what you described as your initial implementation, using an empty object as the source. I might update this to just do that.